zetaclient/chains/solana/observer/inbound.go

```diff
@@ -230,6 +231,33 @@ func FilterInboundEvents(
 		} else {
 			logger.Warn().Msgf("FilterInboundEvents: multiple SPL deposits detected in sig %s instruction %d", tx.Signatures[0], i)
 		}
+
+		// try parsing the instruction as a 'call' if not seen yet
+		if !seenCall {
```

Comment:
It already existed before this PR but why do we use this boolean check instead of using a break in the iteration?